pandas/_libs/missing.pyx

```diff
@@ -160,7 +161,7 @@ def isnaobj_old(arr: ndarray) -> ndarray:
     result = np.zeros(n, dtype=np.uint8)
     for i in range(n):
         val = arr[i]
-        result[i] = val is NaT or _check_none_nan_inf_neginf(val)
```

Comment:
The `_check_none_nan_inf_neginf` helper is only used here I think, so can then be removed